Modules/Tracker/TrackerLinePool.lua

```diff
@@ -0,0 +1,899 @@
+                else
+                    self.texture:SetTexture("Interface\\Buttons\\UI-PlusButton-Up")
+                end
+                self:SetWidth(Questie.db.global.trackerFontSizeQuest + 3)
+                self:SetHeight(Questie.db.global.trackerFontSizeQuest + 3)
+            end
+        end
+
+        expandQuest:SetMode(1) -- maximized
+        expandQuest:EnableMouse(true)
+        expandQuest:RegisterForClicks("LeftButtonUp", "RightButtonUp")
+
+        expandQuest:SetScript("OnClick", function(self)
+            if InCombatLockdown() then
+                return
+            end
+            if self.mode == 1 then
+                self:SetMode(0)
+            else
+                self:SetMode(1)
+            end
+            if Questie.db.char.collapsedQuests[self.questId] then
+                Questie.db.char.collapsedQuests[self.questId] = nil
+            else
+                Questie.db.char.collapsedQuests[self.questId] = true
+            end
+            QuestieTracker:Update()
+        end)
+
+        if Questie.IsWotlk then
+            line:HookScript("OnUpdate", line.OnUpdate)
+        end
+
+        if Questie.db.global.trackerFadeMinMaxButtons then
+            expandQuest:SetAlpha(0)
+        end
+
+        expandQuest:SetScript("OnEnter", function(self)
+            TrackerLinePool.OnHighlightEnter(self)
+            TrackerFadeTicker.OnEnter()
+        end)
+
+        expandQuest:SetScript("OnLeave", function(self)
+            TrackerLinePool.OnHighlightLeave(self)
+            TrackerFadeTicker.OnLeave()
+        end)
+
+        expandQuest:Hide()
+
+        line.expandQuest = expandQuest
+        linePool[i] = line
+        nextFrame = line
+    end
+
+    -- create buttonPool for quest items
+    for i = 1, C_QuestLog.GetMaxNumQuestsCanAccept() do
+        local buttonName = "Questie_ItemButton"..i
+        local btn = CreateFrame("Button", buttonName, UIParent, "SecureActionButtonTemplate")
+        local cooldown = CreateFrame("Cooldown", nil, btn, "CooldownFrameTemplate")
+        btn.range = btn:CreateFontString(nil, "OVERLAY", "NumberFontNormalSmallGray")
+        btn.count = btn:CreateFontString(nil, "ARTWORK", "Game10Font_o1")
+        btn:SetAttribute("type1", "item")
+        btn:SetAttribute("type2", "stop")
+        btn:Hide()
+
+        if Questie.db.global.trackerFadeQuestItemButtons then
+            btn:SetAlpha(0)
+        end
+
+        btn.SetItem = function(self, quest, buttonType, size)
+            local validTexture
+
+            for bag = 0 , 4 do
+                for slot = 1, QuestieCompat.GetContainerNumSlots(bag) do
+                    local texture, _, _, _, _, _, _, _, _, itemID = QuestieCompat.GetContainerItemInfo(bag, slot)
+                    -- These type of quest items can never be secondary buttons
+                    if quest.sourceItemId == itemID and QuestieDB:GetItem(itemID).class == 12 and buttonType == "primary" then
```

Comment:
You should probably use querysingle here rather than getting the whole object (same with the ones below)